### PR TITLE
Stop Banking process, when find out the first result

### PR DESCRIPTION
**Pull request by TruongQuangSB**
- The find Banking process will stop, when find out the first relevant result

### CODE REVIEW

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/ToolboxConfiguration.java [hunk 1]
```diff
@@ -166,7 +166,7 @@ public static boolean isDebugMode() {
 	private static final String BASE_ZOOM_LEVEL = "20"; //$NON-NLS-1$
 	private static final String DEFAULT_COLLISIONS_ENABLED = "true"; //$NON-NLS-1$
 	private static final String DEFAULT_SHEETCUT_CRS = "DR0"; //$NON-NLS-1$
-	private static final String BANK_LINE_TOP_PATH_OFFSET_LIMIT = "1"; //$NON-NLS-1$
+	private static final String BANK_LINE_TOP_PATH_OFFSET_LIMIT = "0.2"; //$NON-NLS-1$
```
Comment (TruongQuangSB) [inline java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/ToolboxConfiguration.java:169]: we should reduce offset here for have the best result

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java [hunk 2]
```diff
@@ -66,4 +75,169 @@ public static Graph<Node, DirectedTOPEdge<Edge>> asDirectedTopGraph(
...
+				}
+				if (edgesDistance
+						.computeIfAbsent(edge,
+								t -> Integer.valueOf(childDistance))
+						.intValue() > childDistance) {
+					edgesDistance.put(edge, Integer.valueOf(childDistance));
+				}
+
+				// Mark the edge's source vertex if needed
+				final Node edgeSource = graph.getEdgeSource(edge);
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java:230]: In which case would the source vertex be needed?

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/DirectedPath.java [hunk 3]
```diff
@@ -0,0 +1,260 @@
...
+	 *
+	 * @return the corresponding GraphPath
+	 */
+	private static GraphPath<Node, DirectedTOPEdge<Edge>> makePath(
+			final Graph<Node, DirectedTOPEdge<Edge>> directedGraph,
+			final List<DirectedTOPEdge<Edge>> edges, final Node startNode) {
+		if (edges.isEmpty()) {
+			return GraphWalk.singletonWalk(directedGraph, startNode, 0);
+		}
+		final Node source = directedGraph.getEdgeSource(edges.get(0));
```
Comment (pstuecker) [inline java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/DirectedPath.java:254]:
```suggestion
		final Node source = directedGraph.getEdgeSource(edges.getFirst());
```

**Review by pstuecker — CHANGES_REQUESTED** [review]

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/DirectedPath.java [hunk 4]
```diff
@@ -0,0 +1,260 @@
...
+import org.eclipse.set.utils.graph.AsSplitTopGraph.Node;
+import org.jgrapht.Graph;
+import org.jgrapht.GraphPath;
+import org.jgrapht.graph.GraphWalk;
+
+/**
+ * Helper class to find directed path between two Node
+ * 
+ */
+public class DirectedPath {
```
[the remark this answers is not in the document]
Comment (TruongQuangSB) [reply java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/DirectedPath.java:32]: ok

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java [hunk 5]
```diff
@@ -66,4 +75,169 @@ public static Graph<Node, DirectedTOPEdge<Edge>> asDirectedTopGraph(
...
+			final int currentDistance = nodeWithDistanceFromEnd.get(vertex)
+					.intValue();
+
+			// Check whether the incoming edges of this node are correctly
+			// decorated
+			for (final DirectedTOPEdge<Edge> edge : graph
+					.incomingEdgesOf(vertex)) {
+				// Mark the edge if needed
+				final BigDecimal weight = edge.edge().getWeight();
+				final int childDistance = weight
```
[the remark this answers is not in the document]
Comment (TruongQuangSB) [reply java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java:216]: ur right, so is better

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java (hunk 2 above)
Comment (TruongQuangSB) [reply java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java:230]: when the distance from target Vertex to this source vertex < max path weight, then mark this source vertex and continue find the relevant edges from this source vertex

File: java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java (hunk 5 above)
[the remark this answers is not in the document]
Comment (TruongQuangSB) [reply java/bundles/org.eclipse.set.utils/src/org/eclipse/set/utils/graph/AsDirectedTopGraph.java:216]: ur right, so is better

**Review by pstuecker — APPROVED** [review]

**Pull request merged**